### PR TITLE
Frontend test

### PR DESCRIPTION
**Pull request by IanChong1**
added weekly view

### CODE REVIEW

File: lib/view/taskView.dart [hunk 1]
```diff
@@ -221,6 +170,59 @@ class _taskViewState extends State<taskView> {
     );
   }
 
+  void _showTaskDetailsDialog(List<Task> tasks) {
```
Comment (Baron-Paelen) [inline lib/view/taskView.dart:173]: is this the same as before, just moved?

File: lib/view/taskView.dart [hunk 2]
```diff
@@ -221,6 +170,59 @@ class _taskViewState extends State<taskView> {
...
+    );
+  }
+
+  void _showTaskNotFoundDialog() {
+    showDialog(
+      context: scaffoldKey.currentState!.context,
+      builder: (BuildContext context) {
+        return AlertDialog(
+          title: const Text('Task Not Found'),
+          content: const Text('The task with ID was not found.'),
```
Comment (Baron-Paelen) [inline lib/view/taskView.dart:212]: maybe add the ID that was searched in here so its more useful

File: lib/view/weeklyTaskView.dart [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    DateTime today = DateTime.now();
+    DateTime start = DateTime(today.year, today.month, today.day);
+    DateTime end = start.add(Duration(days: 7));
+
+    Map<DateTime, List<Task>> activeMap, delayedMap, completedMap;
+
+    // Fetch task maps for the specified week
+    (activeMap, delayedMap, completedMap) = await _db.getTaskMaps(start, end);
+
+    List<Task> allTasks = [
```
Comment (Baron-Paelen) [inline lib/view/weeklyTaskView.dart:45]: This bit is the same as lines 38-42 as taskView, can we abstract it somehow to reduce duplication? They'll probably look for that, or at the very least we can write about how we tried to keep it to a minimum in our final essays.

**Review by Baron-Paelen — APPROVED** [review]

File: lib/view/taskView.dart (hunk 1 above)
Comment (IanChong1) [reply lib/view/taskView.dart:173]: yes

File: lib/view/taskView.dart (hunk 2 above)
Comment (IanChong1) [reply lib/view/taskView.dart:212]: will do mr BARON-PAELEN

File: lib/view/weeklyTaskView.dart (hunk 3 above)
Comment (IanChong1) [reply lib/view/weeklyTaskView.dart:45]: hmmmmmmmmmmmmm, let me try to figure it out.

**Review by lxu03 — APPROVED** [review]

File: lib/view/weeklyTaskView.dart (hunk 3 above)
Comment (stxue1) [reply lib/view/weeklyTaskView.dart:45]: Maybe we should make an equivalent function in the db? Since it looks like you want it formatted in a unique way

**Review by stxue1 — APPROVED** [review]
Lgtm, ill try running it locally later

**Comment by stxue1** [comment]
the user page still says it's you on my screen, is the code to pull the user information hardcoded?

**Comment by anlhu** [comment]
@IanChong1 what do you think about having horizontal divider between days instead of empty days? Just a thought

image

image

**Comment by anlhu** [comment]
image
I am having trouble swiping to weekly view when this is full. do you have any ideas?

File: lib/view/weeklyTaskView.dart (hunk 3 above)
Comment (anlhu) [reply lib/view/weeklyTaskView.dart:45]: You can use the new db function to get a 1 week

File: lib/view/weeklyTaskView.dart [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    return allTasks;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    DateTime today = DateTime.now();
+    List<Widget> dayWidgets = [];
+
+    for (int i = 0; i < 7; i++) {
+      DateTime currentDate = today.add(Duration(days: i));
```
Comment (anlhu) [inline lib/view/weeklyTaskView.dart:62]: I found this way of getting the next day breaks with stuff like daylight savings. I made a function for this in getDateOnly with an offset

File: lib/view/weeklyTaskView.dart [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+      DateTime currentDay =
+          DateTime(currentDate.year, currentDate.month, currentDate.day);
+
+      // Use a Set to store unique tasks for each day
+      Set<Task> uniqueTasksForDay = Set<Task>();
+
+      // Filter tasks for the current day and add them to the Set
+      _allTasks.where((task) {
+        DateTime taskDay = DateTime(task.timeCurrent.year,
```
Comment (anlhu) [inline lib/view/weeklyTaskView.dart:71]: can refactor using getDateOnly

Comment (anlhu) [reply lib/view/weeklyTaskView.dart:71]: I just became THAT guy on the software team 😔

Comment (anlhu) [reply lib/view/weeklyTaskView.dart:71]: But I saw you have a few places where you can remove this duplicated code

**Review by anlhu — CHANGES_REQUESTED** [review]

File: lib/view/weeklyTaskView.dart (hunk 5 above)
Comment (Baron-Paelen) [reply lib/view/weeklyTaskView.dart:71]: can't believe you're THAT guy 🙄

**Pull request merged**